src/group_gui.cpp

```diff
@@ -892,7 +892,12 @@ class VehicleGroupWindow : public BaseVehicleListWindow {
 						/* We do not support VehicleClicked() here since the contextual action may only make sense for individual vehicles */
 
 						if (vindex == v->index) {
-							ShowVehicleListWindow(v);
+							if (vehgroup.NumVehicles() == 1) {
+								ShowVehicleViewWindow(v);
+							}
+							else {
```

Comment:
Oh wait, codestyle - this and below should be on one line

Reply:
Fixed.